### PR TITLE
Test(eos_cli_config_gen): Added test in molecule for `monitor_connectivity` to improve the coverage

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added test in molecule for `monitor_connectivity` to improve the coverage

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Added test in molecule for `monitor_connectivity` to improve the coverage

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 1]
```diff
@@ -285,6 +285,12 @@ As of AVD 5.0.0 "null" option for esp integrity and encryption has been replaced
 +      encryption: disabled
 ```
 
+### `monitor_connectivity.address_only`, `monitor_connectivity.hosts[].address_only`, `monitor_connectivity.vrfs[].address_only` and `monitor_connectivity.vrfs[].hosts[].address_only` default options has been replaced with defined
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
### `address_only` options under `monitor_connectivity` no longer have a default value of `true`
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -285,6 +285,12 @@ As of AVD 5.0.0 "null" option for esp integrity and encryption has been replaced
...
+
+In AVD 4.0.0  the `monitor_connectivity.address_only`, `monitor_connectivity.hosts[].address_only`, `monitor_connectivity.vrfs[].address_only` and `monitor_connectivity.vrfs[].hosts[].address_only` options are default as true so if not defined then it always renders the address-only configuration to set the default interface for probing.
+
+As of AVD 5.0.0 we have to define the `monitor_connectivity.address_only`, `monitor_connectivity.hosts[].address_only`, `monitor_connectivity.vrfs[].address_only` and `monitor_connectivity.vrfs[].hosts[].address_only` options to renders the address-only configuration to set the default interface for probing.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: The suggestion tool is not able to provide the closing tripple backticks, so add that manually.

```suggestion
The default value `true` of data models `monitor_connectivity.hosts[].address_only`, `monitor_connectivity.vrfs[].address_only` and `monitor_connectivity.vrfs[].hosts[].address_only` has been removed, so it is now required to set `address_only: true` to render `address-only` in the EOS configuration.

```diff
 monitor_connectivity:
   hosts:
     - name: server1
       description: server1_connectivity_monitor
       ip: 10.10.10.1
       local_interfaces: HOST_SET
+      address_only: true
       url: https://server1.local.com
   vrfs:
     - name: red
+      address_only: true
       ...
     - name: blue
       ...
       hosts:
         - name: server4
           description: server4_connectivity_monitor
           ip: 10.10.20.1
+          address_only: true
           local_interfaces: VRF_HOST_SET
           url: https://server2.local.com
```

```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/monitor-connectivity.md [hunk 3]
```diff
@@ -98,6 +102,13 @@ interface Management1
 | --------- | ----------- | ------------ | --------------------- | ------------ | --- |
 | server2 | server2_connectivity_monitor | 10.10.20.1 | VRF_HOST_SET | True | https://server2.local.com |
 
+#### Vrf yellow Configuration
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/monitor-connectivity.md:105]: why is this added?

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/monitor-connectivity.cfg [hunk 4]
```diff
@@ -38,28 +38,40 @@ monitor connectivity
       server3_connectivity_monitor
       local-interfaces HOST_SET
       ip 10.10.10.3
+   !
+   host server4
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/monitor-connectivity.cfg:34]: not sure why this one is added?

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/monitor-connectivity.md (hunk 3 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/monitor-connectivity.md:105]: Help to improve the coverage by not defining some params

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/monitor-connectivity.cfg (hunk 4 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/monitor-connectivity.cfg:34]: same as above to improve the coverage

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 1 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: updated

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 2 above)
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: updated with addition of `monitor_connectivity.address_only`

**Review by ClausHolbechArista — APPROVED** [review]

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by Shivani-gslab — COMMENTED** [review]
.

**Review by Shivani-gslab — CHANGES_REQUESTED** [review]
Please remove `default: true` from schema also

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

**Comment by MaheshGSLAB** [comment]
Need to discuss with @gmuloc

**Review by gmuloc — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/monitor-connectivity.yml [hunk 5]
```diff
@@ -27,6 +27,7 @@ monitor_connectivity:
       ip: 10.10.10.3
       local_interfaces: HOST_SET
       address_only: false
+    - name: server4
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/monitor-connectivity.yml:30]: Hmm probably best to not use the same name in a VRF and outside a VRF for our examples

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/monitor-connectivity.yml:30]: it is supported in EOS cli

**Pull request merged**